chapters/chap03-sec02.tex

```diff
@@ -18,17 +18,97 @@ \subsection{\mimc{} Encryption}\label{instantiation:mkhash:mimc-encryption}
+\end{remark}
+
+It is important to note that the security analysis related to algebraic attacks relies on the fact that the polynomial describing the cipher is dense/full. If this assumption is violated, a more granular security analysis needs to be carried for the setting of interest.
+
+\begin{notebox}
+    For \emph{small exponents and large prime fields} (e.g.~for $\secpar = 128$, $p = 2^{128}$ and $\exponent{} = 3$), we see that the lower bound on the number of rounds is (much) smaller in the context of statistical attacks than in the context of algebraic attacks. As such, we see that in such settings algebraic attacks are much more powerful than statistical attacks. Hence, when instanting $\mimc{}$ with an \emph{small} exponent of the form $2^t \pm 1$, it is crucial to make sure that, even if the resulting map is not an APN, the polynomial describing the cipher remains full/very dense. Importantly, if the setting is changed (e.g.~to use exponents that are ``big'' w.r.t.~the field size) the security analysis proposed in~\cite{albrecht2016mimc} must be changed.
+\end{notebox}
+
+\medskip
+
+In the sections below, and as in ~\cite{albrecht2016mimc}, we will consider exponents of the form $\exponent{} = 2^t-1$ and $\exponent{} = 2^t+1$ where $\gcd(\exponent{}, \rCURVE-1) = 1$ (such that exponentiation by $\exponent{}$ yields a permutation on $\FFx{\rCURVE}$). We note that the term cancellation happening with exponents of the form $\exponent{} = 2^t+1$ does not immediatelly translate to the context where $\mimc{}$ is carried out over prime fields of large odd characteristic. In fact, in the case of $\FFx{\rCURVE}$, where $\rCURVE > \binom{\exponent}{\lfloor \exponent / 2
+  \rfloor}$, polynomials $(x + y)^{\exponent}$ are not sparse. This comes from the \emph{Binomial Theorem}
+\[
+  (x + y)^\exponent = \sum_{i = 0}^{\exponent} \binom{\exponent}{i} x^{i}y^{\exponent - i}
+\]
+and the observation that if $\binom{\exponent}{i} < \rCURVE$ then $\binom{\exponent}{i} \bmod \rCURVE = \binom{\exponent}{i}$, hence ensuring that all the polynomial coefficients are greater than $0$, and that the polynomial is dense.
+
+To achieve a security of $\secpar$, we require that $\rounds \geq \secpar \log_{\exponent}(2)$.
+Importantly, since we use $\mimc{}$ over prime fields $\FFx{\rCURVE}$ which are large (where $\rCURVE$ is the prime characteristic of the scalar field of an elliptic curve group, such that $\ceil{\log_2(\rCURVE)} > \secpar$ (i.e.~longer elements are needed in ECC to resist algebraic attacks such that NFS-based attacks on discrete logs for instance)), then, picking $\rounds = \left\lceil \frac{\log_2 \rCURVE}{\log_2 \exponent{}} \right\rceil > \secpar \log_{\exponent}(2)$ provides a margin of safety on the number of rounds selected to instantiate $\mimc{}$ with desired security level $\secpar$.
```

Comment:
NFS -- although many would know what it is we need to expand the abbreviation when it is used for the first time

Reply:
Sure, I'll unroll this one